### ISSUE TITLE
Clean up identical code in codeclimate

### ISSUE THREAD

**Issue by JonathanXu1** [publiclab/plots2#4224]
There are several codeclimate "smells" as detailed in #3706. This task will focus on cleaning up duplicate code.

To do this, we will need to remove duplicate code at:
- [ ] app/controllers/tag_controller.rb line 221 & line 155
- [ ] doc/app/js/searcher.js 115 & 128 &  89
- [ ] app/assets/javascripts/application/i18n/translations.js 2 &  3
- [ ] app/assets/javascripts/question.js 25 &  32
- [ ] app/controllers/admin_controller.rb 35 &  21
- [ ] app/assets/javascripts/jsdiff.js 128 & 139
- [ ] doc/app/js/navigation.js 46 & 55
- [ ] app/controllers/wiki_controller.rb 413 & 426
- [ ] app/assets/javascripts/dragdrop.js 129 & 89
- [ ] app/api/srch/search.rb 264 & 296
- [ ] app/models/node.rb 666 & 695
- [ ] app/api/srch/search.rb 119 & 204
- [ ] app/api/srch/search.rb 147 & 327
- [ ] app/assets/javascripts/like.js 44 & 28
- [ ] app/models/concerns/node_shared.rb 354 & 69 & & 12
- [ ] app/assets/javascripts/jsdiff.js 152 & 144
- [ ] app/models/concerns/node_shared.rb 182 & 215
- [ ] doc/app/js/searcher.js 161 & 139

Keep in mind that though the code in these areas are similar, you can't simply delete the repeated code. Consider creating a function if a variable is different, and always test before checking off any of these issues. 

This covers all the duplicate code, as far as I am able to find within code climate. Please let me know if there's anything else to change.

**Comment by SidharthBansal** [comment]
https://codein.withgoogle.com/dashboard/tasks/6034222042054656/

**Comment by SidharthBansal** [comment]
AWESOME!!!
You did a lot of work. This needs applause too.

**Comment by SidharthBansal** [comment]
@JonathanXu1 is also working on many other parts other than modals. Its great to see his enthusiasm

**Comment by ValentinaTironi** [comment]
Wow, this is amazing 💯 Please, don't forget adding the correspondent labels :)

**Comment by gauravano** [comment]
Hi @JonathanXu1 @ValentinaTironi you want to work on this task? Thanks!

**Comment by ValentinaTironi** [comment]
Yes I think I can do it but in a few days. Today is my birthday 🎁 and I would like enjoying with my family and friends and later is christmas so, can I do it the next week or something like that? Sorry for inconveniences :) happy to help! Thank you!
PD: I’ll try to look the code and think about how to resolve it :)

**Comment by SidharthBansal** [comment]
Happy birthday!!!
Have a wonderful birthday :balloon:

**Comment by gauravano** [comment]
Wishing you a very happy birthday :birthday: :tada: ! 
Always be happy, keep smiling and remain awesome as you are :smiley: .

yes, this issue can wait for sure. Thanks!

**Comment by JonathanXu1** [comment]
Happy birthday, you deserve a break! When you want to work on it, I'd love to help out if you like

**Comment by ValentinaTironi** [comment]
Thank you so much! I’m 18 now hahaha thank you for understanding :) Merry christmas too! 🎄✨

**Comment by ValentinaTironi** [comment]
Hey! I'm ready to work on this. What do you say @JonathanXu1? I'll check the code now and help as I can :)

**Comment by ValentinaTironi** [comment]
I was thinking and I will make different pull requests where the controllers is separated from models, javascripts, etc. Do you agree? @gauravano @JonathanXu1 @SidharthBansal

**Comment by jywarren** [comment]
I think breaking it up into pieces is a great idea. Makes it easier to
review as well!

On Dec 26, 2018 4:45 PM, "Valentina Tironi" <notifications@github.com>
wrote:

I was thinking and I will make different pull requests where the
controllers is separated from models, javascripts, etc. Do you agree?
@gauravano <https://github.com/gauravano> @JonathanXu1
<https://github.com/JonathanXu1> @SidharthBansal
<https://github.com/SidharthBansal>

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/publiclab/plots2/issues/4224#issuecomment-450030794>,
or mute the thread

**Comment by ValentinaTironi** [comment]
Great! I'm on that :)

**Comment by JonathanXu1** [comment]
Alright, how should we divvy up the tasks?

**Comment by ValentinaTironi** [comment]
I'm working now in controllers, maybe you can start with javascripts files? What do you think?

**Comment by JonathanXu1** [comment]
Sure let's do that for now and see

**Comment by ValentinaTironi** [comment]
Great! Let me know if you need help :) Here is my pull request if you check out it #4417 but I'm still working on that so I didn't finish

**Comment by ValentinaTironi** [comment]
OK, I think I finished with controllers so tomorrow I'll starting work with models :)
